### PR TITLE
Lite: Only requires NEON_2_SSE on x86 CPUs (Support Apple Silicon)

### PR DESCRIPTION
**Pull request by prince-chrismc**
This is to support building on M1 Macs. ARM CPU provide these instructions so there no need for the translation layer.

This var can be set when building https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_PROCESSOR.html
or it's provided if You use this build in https://gitlab.kitware.com/cmake/cmake/-/blob/v3.15.7/Modules/CMakeDetermineSystem.cmake#L35

Regardless to have the least impact if it's not set, the current behavoir is mainted and the dependency is pulled in.

This code change does assume it's set correctly

***

I've been working on sharing this project in ConanCenter as I am a community member and discovered this was a patch that the community added some time ago and I want to contribute it back!

### CODE REVIEW

**Comment by gbaned** [comment]
Hi @terryheo Can you please review this PR ? Thank you!

**Review by terryheo — APPROVED** [review]
LGTM

File: tensorflow/lite/CMakeLists.txt [hunk 1]
```diff
@@ -138,14 +138,19 @@ macro(populate_tf_source_vars RELATIVE_DIR SOURCES_VAR)
...
 find_package(Eigen3 REQUIRED)
 find_package(farmhash REQUIRED)
 find_package(fft2d REQUIRED)
 find_package(Flatbuffers REQUIRED)
 find_package(gemmlowp REQUIRED)
-find_package(NEON_2_SSE REQUIRED)
+if (NOT CMAKE_SYSTEM_PROCESSOR OR CMAKE_SYSTEM_PROCESSOR MATCHES "x86")
+  find_package(NEON_2_SSE REQUIRED)
+  list(APPEND TFLITE_TARGET_DEPENDENCIES NEON_2_SSE::NEON_2_SSE)
++endif()
```
Comment (terryheo) [inline tensorflow/lite/CMakeLists.txt:153]: Remove "+"

Comment (prince-chrismc) [reply tensorflow/lite/CMakeLists.txt:153]:
```suggestion
endif()
```

**Review by terryheo — APPROVED** [review]
LGTM

**Comment by gbaned** [comment]
Hi @prince-chrismc Can you please resolve conflicts? Thank you!

**Comment by terryheo** [comment]
> Hi @prince-chrismc Can you please resolve conflicts? Thank you!

I did it.

**Pull request merged**